command/agent/acl_endpoint.go

```diff
@@ -526,3 +524,151 @@ func (s *HTTPServer) aclRoleGetByNameRequest(
 	}
 	return reply.ACLRole, nil
 }
+
+// ACLAuthMethodListRequest performs a listing of ACL auth-methods and is
+// callable via the /v1/acl/auth-methods HTTP API.
+func (s *HTTPServer) ACLAuthMethodListRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+
+	// The endpoint only supports GET requests.
+	if req.Method != http.MethodGet {
+		return nil, CodedError(http.StatusMethodNotAllowed, ErrInvalidMethod)
+	}
+
+	// Set up the request args and parse this to ensure the query options are
+	// set.
+	args := structs.ACLAuthMethodListRequest{}
+
+	if s.parse(resp, req, &args.Region, &args.QueryOptions) {
+		return nil, nil
+	}
+
+	// Perform the RPC request.
+	var reply structs.ACLAuthMethodListResponse
+	if err := s.agent.RPC(structs.ACLListAuthMethodsRPCMethod, &args, &reply); err != nil {
```

Comment:
Oh interesting, is this a new pattern to define RPC methods as constants? I don't think I've seen it before 😄

Reply:
We started doing it with the `ServiceRegistrations` RPC endpoints and I believe all additions since then use this method.